scipy/stats/_hypotests.py

```diff
@@ -663,3 +665,241 @@ def somersd(x, y=None):
+            `pooled` param.
+            ``p_value`` the probability of obtaining a test statistic at least
+            as extreme as the one that was actually observed, assuming that
+            the null hypothesis is true.
+
+    See Also
+    --------
+    chi2_contingency : Chi-square test of independence of variables in a
+        contingency table.
+    fisher_exact : Fisher exact test on a 2x2 contingency table.
+
+    Notes
+    -----
+    Barnard's test is an exact test used in the analysis of contingency
+    tables. It examines the association of two categorical variables and
+    is a more powerful alternative than Fisher's exact test
+    for 2x2 contingency tables.
+    When using Barnard exact test, we can assert three differents null
+    hypothesis :
+    - :math:`H_0 : p_2 \leq p_1`
+    - :math:`H_0 : p_2 \geq p_1`
+    - :math:`H_0 : p_2 = p_1`
+
+    Which can be computed using the ``alternative`` parameter with the
+    values : ``less``, ``greater`` or ``two-sided`` (default one)
+    respectively.
+
+    References
+    ----------
+    .. [1] `Mehta, Cyrus & Senchaudhuri, Pralay. (2003)`_. "Conditional versus Unconditional Exact Tests for Comparing Two Binomials". # noqa: E501
+
+    .. _`Mehta, Cyrus & Senchaudhuri, Pralay. (2003)`: https://www.statsols.com/hubfs/Resources_/Comparing-Two-Binomials.pdf # noqa: E501
+
+    Examples
+    --------
+    An example use of Barnard's test is presented in [1]_.
+
+        Consider the following example of a vaccine efficacy study
+        (Chan, 1998). In a randomized clinical trial of 30 subjects, 15 were
+        innoculated with a recombinant DNA influenza vaccine and the 15 were
+        innoculated with a placebo. Twelve of the 15 subjects in the placebo
+        group (80%) eventually became infected with influenza whereas for the
+        vaccine group, only 7 of the 15 subjects (47%) became infected. The
+        data are tabulated as a 2 x 2 table::
+
+                Vaccine  Placebo
+            Yes     7        12
+            No      8        3
+
+    When working with Statistical hypothesis testing, we usually use a
+    threshold probability or a significance level upon which we decide
+    to reject or not the null hypothesis :math:`H_0`. A commonly used
+    significance level is 5%.
+    To compute the p-value with Banard test, let's call `barnard_exact` as
+    follow :
+
+    >>> import scipy.stats as stats
+    >>> res = stats.barnard_exact([[7, 12], [8, 3]], alternative="less")
+    >>> res.statistic
+    1.894...
+    >>> res.pvalue
+    0.03407...
+
+    Thus, the probability of obtaining test results at least as extreme as the
+    data observed above is 3.4%. In this exemple, we are using the
+    ``alternative`` parameter with the "less" option, because the vaccine has
+    either no effect (:math:`H_0` hypothesis) or a positif effect
+    (:math:`H_1` hypothesis). Therefore, :math:`p_2` is either equal or lower
```

Comment:
p_2 should be defined

Reply:
Update in 9b005f4621ad6895370239958aef1cc6544e08ee, line 772. Sorry, I did multiple changes in one commit. Is this rewording are fine for you?